src/i-shared-minter-mapping.ts

```diff
@@ -0,0 +1,163 @@
+  ProjectCurrencyInfoUpdated,
+  ProjectMaxInvocationsLimitUpdated
+} from "../generated/ISharedMinterV0/ISharedMinterV0";
+
+import {
+  loadOrCreateMinter,
+  generateContractSpecificId,
+  loadOrCreateProjectMinterConfiguration
+} from "./helpers";
+
+///////////////////////////////////////////////////////////////////////////////
+// EVENT HANDLERS start here
+///////////////////////////////////////////////////////////////////////////////
+
+export function handlePricePerTokenInWeiUpdated(
+  event: PricePerTokenInWeiUpdated
+): void {
+  // attempt to load project, if it doesn't exist, log a warning and return
+  // @dev we don't support or allow minters to pre-configure projects that do
+  // not yet exist
+  const project = tryLoadProject(
+    event.params._coreContract,
+    event.params._projectId
+  );
+  if (!project) {
+    log.warning("Project {} not found for core contract {}", [
+      event.params._projectId.toString(),
+      event.params._coreContract.toHexString()
+    ]);
+    return;
+  }
+
+  // load minter
+  const minter = loadOrCreateMinter(event.address, event.block.timestamp);
+
+  // load or create project minter configuration
+  const projectMinterConfig = loadOrCreateProjectMinterConfiguration(
+    project,
+    minter
+  );
+
+  projectMinterConfig.basePrice = event.params._pricePerTokenInWei;
+  projectMinterConfig.priceIsConfigured = true;
+  projectMinterConfig.save();
+
+  // update project's updatedAt timestamp to induce a sync
+  // @dev this may induce a sync if the project's minter is not this minter,
+  // but good to consider the project as "updated", as it could switch to this
+  // minter in the future
+  project.updatedAt = event.block.timestamp;
+  project.save();
+}
+
+export function handleProjectCurrencyInfoUpdated(
+  event: ProjectCurrencyInfoUpdated
+): void {
+  // attempt to load project, if it doesn't exist, log a warning and return
+  // @dev we don't support or allow minters to pre-configure projects that do
+  // not yet exist
+  const project = tryLoadProject(
+    event.params._coreContract,
+    event.params._projectId
+  );
+  if (!project) {
+    log.warning("Project {} not found for core contract {}", [
+      event.params._projectId.toString(),
+      event.params._coreContract.toHexString()
+    ]);
+    return;
+  }
+
+  // load minter
+  const minter = loadOrCreateMinter(event.address, event.block.timestamp);
+
+  // load or create project minter configuration
+  const projectMinterConfig = loadOrCreateProjectMinterConfiguration(
+    project,
+    minter
+  );
+
+  projectMinterConfig.currencyAddress = event.params._currencyAddress;
+  projectMinterConfig.currencySymbol = event.params._currencySymbol;
+  projectMinterConfig.save();
+
+  // update project's updatedAt timestamp to induce a sync
+  // @dev this may induce a sync if the project's minter is not this minter,
+  // but good to consider the project as "updated", as it could switch to this
+  // minter in the future
```

Comment:
Wouldn't we sync the project with the updated minter config once the minter is assigned to the project anyways? Also okay with leaving this here regardless so that our db row for the minter config is always up to date. We should probably consider syncing minter configs separately in the future though assuming we want to sync all changes.

Reply:
Thanks for this comment. It is true that we only induce syncs of the active projectMinterConfig, so I think we should only update project `updatedAt` to induce that sync if the projectMinterConfig is the current project minter configuration.
Updated logic in commit [b89ad8d](https://github.com/ArtBlocks/artblocks-subgraph/pull/229/commits/b89ad8d1e027d00d38cc16b2fe3c59679b65ab3a)